com.oracle.truffle.r.ffi.impl/src/com/oracle/truffle/r/ffi/impl/nodes/IntegerIsSortedNode.java

```diff
@@ -0,0 +1,38 @@
+package com.oracle.truffle.r.ffi.impl.nodes;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.r.runtime.data.RIntVector;
+import com.oracle.truffle.r.runtime.data.altrep.AltrepSortedness;
+import com.oracle.truffle.r.runtime.data.altrep.AltrepUtilities;
+import com.oracle.truffle.r.runtime.ffi.AltrepRFFI;
+
+@GenerateUncached
+public abstract class IntegerIsSortedNode extends FFIUpCallNode.Arg1 {
```

Comment:
The idea is that in such nodes, we should delegate to `VectorDataLibrary#isSorted` (which should delegate to the ALTREP method if appropriate): this way, for example, also our sequences will respond to `INTEGER_IS_SORTED`, which is what we want: support for user defined ALTREP objects, but also our own compact representations should implement this API.

Reply:
So is it okay if I change the return type of `VectorDataLibrary.isSorted` to an enum? Since `INTEGER_IS_SORTED` essentialy returns an enum of 6 possible values (denoted as `AltrepSortedness.java` in my code)